src/graph/Sankey.js

```diff
@@ -0,0 +1,232 @@
+    Sankey.prototype.publish("vertexWidth", 36, "number", "Vertex Width");
+    Sankey.prototype.publish("vertexPadding", 40, "number", "Vertex Padding");
+    Sankey.prototype.publish("xAxisMovement", false, "boolean", "Enable x-axis movement");
+    Sankey.prototype.publish("yAxisMovement", false, "boolean", "Enable y-axis movement");
+
+    Sankey.prototype.sankeyData = function () {
+        var vertexIndex = {};
+        var retVal = {
+            vertices: [],
+            edges: []
+        };
+        var mappings = this.mappings().filter(function (mapping) { return mapping.column(); });
+        mappings.forEach(function (mapping, idx) {
+            var view = this._db.rollupView([mapping.column()]);
+            view.entries().forEach(function (row) {
+                var id = mapping.column() + ":" + idx + ":" + row.key;
+                if (!vertexIndex[id]) {
+                    retVal.vertices.push({
+                        __id: id,
+                        __category: mapping.column(),
+                        name: row.key,
+                        origRow: row
+                    });
+                    vertexIndex[id] = retVal.vertices.length - 1;
+                }
+            }, this);
+        }, this);
+        mappings.forEach(function (mapping, idx) {
+            if (idx < mappings.length - 1) {
+                var mapping2 = mappings[idx + 1];
+                var view = this._db.rollupView([mapping.column(), mapping2.column()]);
+                view.entries().forEach(function (row) {
+                    var sourceID = mapping.column() + ":" + idx + ":" + row.key;
+                    row.values.forEach(function (value) {
+                        var targetID = mapping2.column() + ":" + (idx + 1) + ":" + value.key;
+                        retVal.edges.push({
+                            __id: sourceID + "_" + targetID,
+                            source: vertexIndex[sourceID],
+                            target: vertexIndex[targetID],
+                            value: mapping2.aggregate(value.values)
+                        });
+                    });
+                });
+            }
+        }, this);
+
+        return retVal;
+    };
+
+    Sankey.prototype.enter = function (domNode, element) {
+        SVGWidget.prototype.enter.apply(this, arguments);
+
+        this._d3Sankey = d3.sankey();
+        this._d3SankeyPath = this._d3Sankey.link();
+    };
+
+    Sankey.prototype.update = function (domNode, element) {
+        SVGWidget.prototype.update.apply(this, arguments);
+
+        this._palette = this._palette.switch(this.paletteID());
+
+        var sankeyData = this.sankeyData();
+        this._d3Sankey
+            .size([this.width(), this.height()])
+            .nodeWidth(this.vertexWidth())
+            .nodePadding(this.vertexPadding())
+            .nodes(sankeyData.vertices)
+            .links(sankeyData.edges)
+            .layout(32)
```

Comment:
not sure what layout(32) is but flagging it as having a hard set value being passed in